stl/inc/ranges

```diff
@@ -1328,8 +1328,190 @@ namespace ranges {
+                noexcept(_Iter += _Off)) /* strengthened */ {
+                return _Iter += _Off;
+            }
+
+            _NODISCARD_FRIEND constexpr _Iterator operator-(_Iterator _Iter, difference_type _Off) noexcept(
+                noexcept(_Iter -= _Off)) /* strengthened */ {
+                return _Iter -= _Off;
+            }
+            // clang-format off
+            _NODISCARD_FRIEND constexpr difference_type operator-(const _Iterator& _Left,const _Iterator& _Right)
+                noexcept(noexcept(_Left._Current - _Right._Current)) /* strengthened */ {
+                // clang-format on
+                return static_cast<difference_type>(_Left._Current) - static_cast<difference_type>(_Right._Current);
+            }
+        };
+
+        /* [[no_unique_address]] */ _Movable_box<_Wx> _Value{};
+        /* [[no_unique_address]] */ _Bo _Bound{};
+
+        template <class _Tuple1, class _Tuple2, size_t... _Indexes1, size_t... _Indexes2>
+        _NODISCARD_CTOR constexpr repeat_view(
+            _Tuple1& _Val1, _Tuple2& _Val2, index_sequence<_Indexes1...>, index_sequence<_Indexes2...>) noexcept
+            : _Value(in_place, _Tuple_get<_Indexes1>(_STD move(_Val1))...),
+              _Bound(_Tuple_get<_Indexes2>(_STD move(_Val2))...) {}
+
+    public:
+        // clang-format off
+        repeat_view() requires default_initializable<_Wx> = default;
+
+        _NODISCARD_CTOR constexpr explicit repeat_view(const _Wx& _Value_, _Bo _Bound_ = _Bo())
+            noexcept(is_nothrow_copy_constructible_v<_Wx> && is_nothrow_copy_constructible_v<_Bo>) // strengthened
+            requires copy_constructible<_Wx> : _Value(in_place, _Value_),
+                                               _Bound(_Bound_) {
+            // clang-format on
+            if constexpr (!is_same_v<_Bo, unreachable_sentinel_t>) {
+                _STL_ASSERT(_Bound >= 0, "Bound must be >= 0");
+            }
+        }
+        _NODISCARD_CTOR constexpr explicit repeat_view(_Wx&& _Value_, _Bo _Bound_ = _Bo()) noexcept(
+            is_nothrow_move_constructible_v<_Wx>&& is_nothrow_copy_constructible_v<_Bo>) // strengthened
+            : _Value(in_place, std::move(_Value_)), _Bound(_Bound_) {
+            if constexpr (!is_same_v<_Bo, unreachable_sentinel_t>) {
+                _STL_ASSERT(_Bound >= 0, "Bound must be >= 0");
+            }
+        }
+        template <class... _WArgs, class... _BArgs>
+        requires constructible_from<_Wx, _WArgs...> && constructible_from<_Bo, _BArgs...>
+            _NODISCARD_CTOR constexpr explicit repeat_view(piecewise_construct_t, tuple<_WArgs...> _Val_args,
+                tuple<_BArgs...> _Bound_args = tuple<>{}) noexcept(is_nothrow_constructible_v<_Wx, _WArgs...>&&
+                    is_nothrow_constructible_v<_Bo, _BArgs...>)
+            : repeat_view(_Val_args, _Bound_args, index_sequence_for<_WArgs...>{}, index_sequence_for<_BArgs...>{}) {}

```

Comment:
I wonder whether it would be better to directly construct the `_Bo` value here and then simply pass it to the internal ctor, which can make the internal ctor less templatized.

Reply:
Given that `_Bo` is a type we know (and is always either an integral type or an `Int128`), I think this is just the right thing to do here.

Reply:
Oh, it seems suitable to use `make_from_tuple` to construct the `_Bo` value here, but we haven't strengthened its exception specification yet.

Reply:
So what I have now is incorrect til that's fixed?

Reply:
I think the current usage is OK because compilers can correctly figure out the actually non-throwing cases. Strengthening can be investigated later.